### PR TITLE
More generous timeout in DaemonicSpec

### PR DESCRIPTION
**Pull request by johanandren**
Refs #24162 

Also:
 * allow it to complete faster if possible
 * avoid potential race with port number
 * always shut the extra actor system down

### CODE REVIEW

File: akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala [hunk 1]
```diff
@@ -38,17 +27,23 @@ class DaemonicSpec extends AkkaSpec {
...
 
-      // get new non daemonic threads running
-      val newNonDaemons: Set[Thread] = Thread.getAllStackTraces().keySet().asScala.seq.
-        filter(t ⇒ !origThreads(t) && t.isDaemon == false).to[Set]
+        // get new non daemonic threads running
+        awaitAssert({
+          val newNonDaemons: Set[Thread] = Thread.getAllStackTraces.keySet().asScala.seq.
+            filter(t ⇒ !origThreads(t) && !t.isDaemon).to[Set]
+          newNonDaemons should ===(Set.empty[Thread])
+        }, 4.seconds.dilated)
```
Comment (patriknw) [inline akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala]: that is already dilated by `awaitAssert`

**Review by patriknw — APPROVED** [review]
LGTM after removal of dilated

File: akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala (hunk 1 above)
Comment (johanandren) [reply akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala]: D'oh. I never remember where it is and where it isn't...

File: akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala [hunk 2]
```diff
@@ -38,17 +27,23 @@ class DaemonicSpec extends AkkaSpec {
         akka.log-dead-letters-during-shutdown = off
       """))
 
-      val unusedAddress = getOtherAddress(daemonicSystem, "tcp").copy(port = Some(unusedPort))
-      val selection = daemonicSystem.actorSelection(s"${unusedAddress}/user/SomeActor")
-      selection ! "whatever"
-      Thread.sleep(2.seconds.dilated.toMillis)
+      try {
+        val unusedPort = 86 // very unlikely to ever be used, "system port" range reserved for Micro Focus Cobol
```
Comment (chbatey) [inline akka-remote/src/test/scala/akka/remote/DaemonicSpec.scala]: lol

**Review by chbatey — APPROVED** [review]
LGTM

**Pull request merged**